commands/playwright.js

```diff
@@ -177,89 +306,146 @@ module.exports = {
-    await module.exports.waitUntilStable();
+    if (provider === 'metamask') {
+      await module.exports.waitUntilStable(provider);
+    }
   },
-  async waitAndType(selector, value, page = metamaskWindow) {
+  async waitAndType(
+    provider,
+    selector,
+    value,
+    page = module.exports.windows(provider),
+  ) {
     if (typeof value === 'number') {
       value = value.toString();
     }
-    const element = await module.exports.waitFor(selector, page);
+    const element = await module.exports.waitFor(provider, selector, page);
     await element.type(value);
-    await module.exports.waitUntilStable(page);
+    if (provider === 'metamask') {
+      await module.exports.waitUntilStable(provider, page);
+    }
   },
-  async waitAndGetValue(selector, page = metamaskWindow) {
+  async waitAndGetValue(
+    provider,
+    selector,
+    page = module.exports.windows(provider),
+  ) {
     const expect = global.expect
       ? global.expect
       : require('@playwright/test').expect;
-    const element = await module.exports.waitFor(selector, page);
+    const element = await module.exports.waitFor(provider, selector, page);
     await expect(element).toHaveText(/[a-zA-Z0-9]/, {
       ignoreCase: true,
       useInnerText: true,
     });
     const value = await element.innerText();
     return value;
   },
-  async waitAndGetInputValue(selector, page = metamaskWindow) {
+  async waitAndGetInputValue(
+    provider,
+    selector,
+    page = module.exports.windows(provider),
+  ) {
     const expect = global.expect
       ? global.expect
       : require('@playwright/test').expect;
-    const element = await module.exports.waitFor(selector, page);
+    const element = await module.exports.waitFor(provider, selector, page);
     await expect(element).toHaveValue(/[a-zA-Z1-9]/);
     const value = await element.inputValue();
     return value;
   },
-  async waitAndGetAttributeValue(selector, attribute, page = metamaskWindow) {
+  async waitAndGetAttributeValue(
+    provider,
+    selector,
+    attribute,
+    page = module.exports.windows(provider),
+  ) {
     const expect = global.expect
       ? global.expect
       : require('@playwright/test').expect;
-    const element = await module.exports.waitFor(selector, page);
+    const element = await module.exports.waitFor(provider, selector, page);
     await expect(element).toHaveAttribute(attribute, /[a-zA-Z0-9]/);
     const attrValue = await element.getAttribute(attribute);
     return attrValue;
   },
-  async waitAndSetValue(text, selector, page = metamaskWindow) {
-    const element = await module.exports.waitFor(selector, page);
+  async waitAndSetValue(
+    provider,
+    text,
+    selector,
+    page = module.exports.windows(provider),
+  ) {
+    const element = await module.exports.waitFor(provider, selector, page);
     await element.fill('');
-    await module.exports.waitUntilStable(page);
+    await module.exports.waitUntilStable(provider, page);
```

Comment:
In some functions, you introduce an if statement that runs `.waitUntillStable()` only if the provider is set to MetaMask, yet sometimes you use this function for Phantom too. What's the deal with it?

Reply:
i found that for phantom it's not always needed, only when there are animations it's nice to have.